{
  try {
    super.runTest();
  }
 catch (  Exception e) {
    if (fExpected.isAssignableFrom(e.getClass()))     return;
 else     throw e;
  }
  fail("Expected exception " + fExpected);
}
